src/base/private/profile_p.cpp

```diff
@@ -76,9 +76,15 @@ QString Private::DefaultProfile::dataLocation() const
 #if defined(Q_OS_WIN) || defined (Q_OS_MACOS)
     return locationWithConfigurationName(QStandardPaths::AppLocalDataLocation);
 #else
-    // on Linux gods know why qBittorrent creates 'data' subdirectory in ~/.local/share/
-    return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
+    // On Linux use legacy ~/.local/share/data/ if it exists
+    const QString legacydir = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
         + QLatin1String("/data/") + profileName() + QLatin1Char('/');
+
+    if (QDir(legacydir).exists())
+        return legacydir;
+
+    return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
+        + QLatin1Char('/') + profileName() + QLatin1Char('/');
```

Comment:
I think I single return point would be better, and you should log a warning or something. For example: `"The legacy data directory is being used (" + legacyDir + "). It is recommended to move the contents of that folder to " + configDir`.

```cpp
    // On Linux use legacy ~/.local/share/data/ if it exists
    const QString configDir = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
        + QLatin1Char('/') + profileName() + QLatin1Char('/');
    const QString legacyDir = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
        + QLatin1String("/data/") + profileName() + QLatin1Char('/');

    if (QDir(legacyDir).exists()) {
        configDir = legacyDir;
        // log something at warning level about this...
    }

    return configDIr;
```